barretenberg/cpp/src/barretenberg/ecc/fields/field_declarations.hpp

```diff
@@ -89,6 +89,16 @@ template <class Params_> struct alignas(32) field {
     constexpr field(const uint64_t a, const uint64_t b, const uint64_t c, const uint64_t d) noexcept
         : data{ a, b, c, d } {};
 
+    constexpr explicit field(const uint512_t& input) noexcept
+    {
+        uint256_t value = (input % modulus).lo;
```

Comment:
Note to self: Assuming the modulus is < 256 bits, .lo will represent the result of the modulo. 

Going to come back to this, to check usage

Reply:
I added a comment that describes intended usage